extensions/wikia/AchievementsII/services/AchAwardingService.class.php

```diff
@@ -215,30 +215,30 @@ private function calculateAndSaveScore() {
 			// notes for later refactoring
 			// what do I need here?
 			// - number of points based on level - give level, get points
-			foreach ( $this->mBadges as $badge_type_id => $badge_laps ) {
+			foreach( $this->mBadges as $badge_type_id => $badge_laps ) {
 
 				$badgeType = AchConfig::getInstance()->getBadgeType( $badge_type_id );
 
-				if ( $badgeType == BADGE_TYPE_NOTINTRACKSTATIC ) {
+				if( $badgeType == BADGE_TYPE_NOTINTRACKSTATIC ) {
 
-					$score += AchConfig::getInstance()->getLevelScore( $notInTrackStatic[$badge_type_id]['level'] ) * ( ( AchConfig::getInstance()->isInfinite( $badge_type_id ) ) ? $badge_laps : 1 );
+					$score += AchConfig::getInstance()->getLevelScore( $notInTrackStatic[ $badge_type_id ][ 'level' ] ) * ( ( AchConfig::getInstance()->isInfinite( $badge_type_id ) ) ? $badge_laps : 1 );
 
-				} else if ( $badgeType == BADGE_TYPE_NOTINTRACKCOMMUNITYPLATINUM ) {
+				} else if( $badgeType == BADGE_TYPE_NOTINTRACKCOMMUNITYPLATINUM ) {
 
 					$score += AchConfig::getInstance()->getLevelScore( BADGE_LEVEL_PLATINUM );
 
-				} else if ( $badgeType == BADGE_TYPE_INTRACKSTATIC || $badgeType == BADGE_TYPE_INTRACKEDITPLUSCATEGORY ) {
+				} else if( $badgeType == BADGE_TYPE_INTRACKSTATIC || $badgeType == BADGE_TYPE_INTRACKEDITPLUSCATEGORY ) {
 
-					if ( $badgeType == BADGE_TYPE_INTRACKEDITPLUSCATEGORY ) {
+					if( $badgeType == BADGE_TYPE_INTRACKEDITPLUSCATEGORY ) {
 						$badge_type_id = BADGE_EDIT;
 					}
 
-					$maxPoints = AchConfig::getInstance()->getLevelScore( $inTrackStatic[$badge_type_id]['laps'][count( $inTrackStatic[$badge_type_id]['laps'] )-1]['level'] );
+					$maxPoints = AchConfig::getInstance()->getLevelScore( $inTrackStatic[ $badge_type_id ][ 'laps' ][ count( $inTrackStatic[ $badge_type_id ][ 'laps' ] ) - 1 ][ 'level' ] );
 
-					foreach ( $badge_laps as $badge_lap ) {
+					foreach( $badge_laps as $badge_lap ) {
 
-						if ( isset( $inTrackStatic[$badge_type_id]['laps'][$badge_lap] ) ) {
-							$score += AchConfig::getInstance()->getLevelScore( $inTrackStatic[$badge_type_id]['laps'][$badge_lap]['level'] );
+						if( isset( $inTrackStatic[ $badge_type_id ][ 'laps' ][ $badge_lap ] ) ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)